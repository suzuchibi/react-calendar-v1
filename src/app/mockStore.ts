import configureMockStore from 'redux-mock-store';
const mock = configureMockStore();

export const mockStore = mock({
  drawer: false,
  today: 1626015600000,
  modal: {
    isActive: false,
    mode: '0',
  },
  form: {
    id: '0',
    isDuplicate: false,
  },
  cal: {
    current: 1625065200000,
    holidays: [
      {
        date: '2021-07-22',
        summary: '海の日',
      },
    ],
    status: 'loaded',
  },
  calPoper: {
    anchorEl: null,
  },
  schedule: {
    holidays: [],
    selected: 1626015600000,
    status: 'loaded',
  },
  datas: [
    {
      id: '1',
      title: 'タイトル1',
      coment: 'コメント1',
      start: 1626139200000,
      end: 1626141600000,
    },
  ],
  screen: false,
  snack: {
    open: false,
    msg: '',
  },
});
