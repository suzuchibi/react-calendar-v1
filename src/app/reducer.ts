import { combineReducers } from 'redux';
import isDrawerReducer from '../features/side/isDrawerReducer';
import todayReducer from '../features/today/todayReducer';
import modalReducer from '../features/modal/reducer';
import formReducer from '../features/form/reducer';
import calendarReducer from '../features/calendar/calendarReducer';
import calendarPoperReducer from '../features/dialog/calendarPoper/reducer';
import scheduleReducer from '../features/scheduler/reducers/scheduleReducer';
import scheduleDatasReducer from '../features/scheduler/reducers/datasReducer';
import fullScreenDialogReducer from '../features/dialog/fullScreen/reducer';
import snackReducer from '../features/snack/snackReducer';

const rootReducer = combineReducers({
  drawer: isDrawerReducer,
  today: todayReducer,
  modal: modalReducer,
  form: formReducer,
  cal: calendarReducer,
  calPoper: calendarPoperReducer,
  schedule: scheduleReducer,
  datas: scheduleDatasReducer,
  screen: fullScreenDialogReducer,
  snack: snackReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
