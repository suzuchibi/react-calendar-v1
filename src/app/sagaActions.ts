import { Holidays } from '../features/calendar/types';

const GOOGLE_CALENDAR_API = 'app/googleCalendarAPI';
const GOOGLE_CALENDAR_API_SUCESS = 'app/googleCalendarApiSuccess';
const GOOGLE_CALENDAR_API_ERROR = 'app/googleCalendarApiError';

export interface Props {
  mode: string;
  stamp: number;
}

export interface GoogleCalendarApiProps {
  type: typeof GOOGLE_CALENDAR_API;
  payload: Props;
}

interface HolidayProps {
  mode: string;
  result: Array<Holidays>;
}

export interface GoogleCalendarApiSuccessProps {
  type: typeof GOOGLE_CALENDAR_API_SUCESS;
  payload: HolidayProps;
}

export interface GoogleCalendarApiErrorProps {
  type: typeof GOOGLE_CALENDAR_API_ERROR;
  payload: {
    mode: string;
  };
}

export type ActionTypes =
  | GoogleCalendarApiProps
  | GoogleCalendarApiSuccessProps
  | GoogleCalendarApiErrorProps;

/**
 * google calendar API saga start
 * @param {object} payload
 */
export const gApiSaga = (payload: Props): ActionTypes => ({
  type: GOOGLE_CALENDAR_API,
  payload,
});

/**
 * Success
 * @param {object} payload
 */
export const gApiSuccess = (payload: HolidayProps): ActionTypes => ({
  type: GOOGLE_CALENDAR_API_SUCESS,
  payload,
});

/**
 * Error
 * @param {object} payload
 */
export const gApiError = (payload: { mode: string }): ActionTypes => ({
  type: GOOGLE_CALENDAR_API_ERROR,
  payload,
});
