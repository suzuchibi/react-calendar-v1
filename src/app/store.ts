import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas';
import rootReducer from './reducer';

const saga = createSagaMiddleware();
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(saga))
);
saga.run(rootSaga);

export default store;
