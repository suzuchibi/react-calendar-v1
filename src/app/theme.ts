import { createMuiTheme } from '@material-ui/core/styles';

export const theme = createMuiTheme({
  palette: {
    background: {
      default: '#fff',
    },
  },
});
