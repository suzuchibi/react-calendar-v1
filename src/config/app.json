{
  "drawerWidth": 260,
  "timeLineWidth": 60,
  "startOrder": 10,
  "lastOrder": 19,
  "dayOfTheWeek": ["日", "月", "火", "水", "木", "金", "土"],
  "gapiCal": {
    "key": "AIzaSyCLqi5tQwWw8qEGAWc9NC1cEU78C4EAWkM",
    "address": "ja.japanese#holiday@group.v.calendar.google.com",
    "gurl": "https://www.googleapis.com/calendar/v3/calendars/"
  }
}