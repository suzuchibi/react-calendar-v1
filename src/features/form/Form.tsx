import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { RootState } from '../../app/reducer';
import { Datas } from '../scheduler/types';
import classNames from 'classnames';
import ErrorMsg from './ErrorMsg';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { makeStyles } from '@material-ui/core/styles';
import CreateIcon from '@material-ui/icons/Create';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Checkbox from '@material-ui/core/Checkbox';
import Button from '@material-ui/core/Button';
import teal from '@material-ui/core/colors/teal';
import {
  add,
  getYear,
  getMonth,
  getDate,
  getHours,
  getMinutes,
} from 'date-fns';
import validator, { config as setting, DatasTypesValidator } from './validator';

// Actions
import { handleDuplicate } from './actions';
import { handleDrawer } from '../side/isDrawerReducer';
import { handleIsActive } from '../modal/actions';
import {
  postAdded,
  postUpdated,
  postRemoved,
} from '../scheduler/actions/datasActions';
import { handleFullScreenDialog } from '../dialog/fullScreen/reducer';

/**
 * CSS Styling
 * @return {Object}
 */
const useStyles = makeStyles(() => ({
  root: {
    '& .MuiOutlinedInput-multiline.MuiOutlinedInput-marginDense': {
      padding: '8px',
    },
  },
  titleBox: {
    display: 'flex',
    alignItems: 'center',
    margin: '0 0 8px 0',
  },
  reset: {
    margin: 0,
    padding: 0,
  },
  marginRight: {
    marginRight: '16px',
  },
  marginBottom: {
    marginBottom: '24px',
  },
  formFlex: {
    display: 'flex',
    flexDirection: 'column',
  },
  formTitle: {
    color: '#666',
    fontSize: '12px',
  },
  formTimeFlex: {
    display: 'flex',
  },
  formTextAreaTitle: {
    color: '#666',
    fontSize: '12px',
    marginBottom: '6px',
  },
  formTextArea: {
    width: '100%',
    fontSize: '14px',
    borderColor: '#aaa',
    letterSpacing: '0.8px',
    margin: '0 0 16px 0',
    padding: 0,
  },
  formCheckBox: {
    marginRight: '4px',
    padding: 0,
  },
  errorText: {
    fontSize: '12px',
    margin: 0,
    padding: 0,
  },
  deleteText: {
    paddingTop: '6px',
    fontSize: '12px',
  },
  buttonArea: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  okButton: {
    color: '#fff',
    background: teal[500],
    '&:hover': {
      background: teal[600],
    },
    marginRight: '16px',
  },
}));

/**
 * Select Datas By Id
 * @param {Array} datas
 * @param {String} id
 */
const selectDatasById = (datas: Datas, id: String) =>
  datas.find((f) => f.id === id);

/**
 * Init Defaults Setting to ADD
 * @param {number} timeStamp
 * @return {Object}
 */
const initAddDefaults = (timeStamp: number) => {
  const current = getHours(new Date());
  const defaults = {
    start: setting.defaultTime.start,
    end: setting.defaultTime.end,
  };
  /**
   * @return {Number}
   */
  const startTime = () => {
    if (current < defaults.start || defaults.end < current)
      return defaults.start;
    return Number(current);
  };
  /**
   * @return {String}
   */
  const dateTime = () => {
    const dates = {
      year: getYear(timeStamp),
      month: ('00' + (getMonth(timeStamp) + 1)).slice(-2),
      date: ('00' + getDate(timeStamp)).slice(-2),
    };
    if (defaults.end < current) {
      const added = add(timeStamp, { days: 1 });
      dates.year = getYear(added);
      dates.month = ('00' + (getMonth(added) + 1)).slice(-2);
      dates.date = ('00' + getDate(added)).slice(-2);
    }
    return `${dates.year}-${dates.month}-${dates.date}`;
  };
  return {
    id: '0',
    title: '',
    date: dateTime(),
    start: `${('00' + startTime()).slice(-2)}:00`,
    end: `${('00' + (startTime() + 1)).slice(-2)}:00`,
    memo: '',
    delete: false,
  };
};

/**
 * Init Defaults Setting to Edit
 * @param {String} id
 * @param {Array} fetch
 * @return {Object}
 */
const initEditDefaults = (id: String, fetch: Datas) => {
  const parse = selectDatasById(fetch, id);
  if (parse) {
    const start = parse.start;
    const end = parse.end;
    const y = getYear(start);
    const m = getMonth(start) + 1;
    const d = getDate(start);
    const sH = getHours(start);
    const sM = getMinutes(start);
    const eH = getHours(end);
    const eM = getMinutes(end);
    return {
      id: id,
      title: parse.title,
      date: `${y}-${('00' + m).slice(-2)}-${('00' + d).slice(-2)}`,
      start: `${('00' + sH).slice(-2)}:${('00' + sM).slice(-2)}`,
      end: `${('00' + eH).slice(-2)}:${('00' + eM).slice(-2)}`,
      memo: parse.coment,
      delete: false,
    };
  }
};

/**
 * Init Defaults Setting Controller
 * @param {String} id
 * @param {number} timeStamp
 * @param {Array} fetch
 * @return {Function}
 */
const initDefaults = (id: String, timeStamp: number, fetch: Datas) => {
  switch (id) {
    case '0':
      return initAddDefaults(timeStamp);
    default:
      return initEditDefaults(id, fetch);
  }
};

/**
 * Init Required
 * @param {Object} forms
 * @return {Object}
 */
const initRequired = (forms: DatasTypesValidator) => {
  return {
    title: forms.title,
    start: forms.start,
    end: forms.end,
  };
};

/**
 * Render
 */
function Form() {
  const classes = useStyles();
  const message = { ...setting.errorResetMessage };
  const md = useMediaQuery('(max-width:960px)');
  // Redux State
  const id = useSelector((state: RootState) => state.form.id);
  const isDrawer = useSelector((state: RootState) => state.drawer);
  const mode = useSelector((state: RootState) => state.modal.mode);
  const fetch = useSelector((state: RootState) => state.datas);
  const selected = useSelector((state: RootState) => state.schedule.selected);
  const defaults = Object.assign({}, initDefaults(id, selected, fetch));
  const isScreen = useSelector((state: RootState) => state.screen);
  const isDupError = useSelector((state: RootState) => state.form.isDuplicate);
  // Local State
  const [forms, setForms] = useState(defaults);
  const [required, setRequired] = useState(initRequired(forms));
  const [errors, setErrors] = useState(message);
  // const [isDuplicateError, setIsDuplicate] = useState(false);
  const [disabled, setDisabled] = useState(false);
  // Actions
  const title = (m: String) => (m === '0' ? 'Add' : 'Edit');
  const label = (m: String) => (m === '0' ? 'add' : 'update');
  const dispatch = useDispatch();
  const close = () => {
    if (!isScreen) {
      dispatch(handleIsActive(false));
    } else {
      dispatch(handleFullScreenDialog(false));
    }
    dispatch(handleDuplicate(false));
  };

  const handleChange = (
    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    const name = e.target.name;
    const value = e.target.value;
    const ignore = setting.ignore.find((v) => name === v);
    setForms({ ...forms, [name]: value });
    if (!ignore) {
      setRequired({ ...required, [name]: value });
      setErrors({
        ...errors,
        [name]: validator.start(name, value),
      });
    }
  };

  const handleCheckBox = (e: React.ChangeEvent<HTMLInputElement>) => {
    const name = e.target.name;
    const value = e.target.checked;
    setForms({ ...forms, [name]: value });
  };

  const isButtonDisabled = () => {
    const isReq = Object.values(required).filter((v) => v === '').length === 0;
    const isError = Object.values(errors).filter((v) => v !== '').length === 0;
    return isReq && isError && !disabled;
  };

  const duplicateChecker = () => {
    return validator.duplicateCheck(fetch, forms.date, forms.start, forms.end);
  };

  const submitButton = () => {
    setDisabled(true);
    const post = Object.assign({}, forms);
    if (isButtonDisabled()) {
      // Duplicate Checke
      if (id === '0' && duplicateChecker()) {
        dispatch(handleDuplicate(true));
        setDisabled(false);
        return;
      }
      switch (id) {
        case '0':
          dispatch(postAdded(post));
          break;
        default:
          post.delete === undefined || post.delete === false
            ? dispatch(postUpdated(post))
            : dispatch(postRemoved(id));
          break;
      }
      if (md && !isDrawer) dispatch(handleDrawer(true));
      close();
      // dispatch(openSuccess());
    }
    setDisabled(false);
  };

  useEffect(() => {
    if (validator.overCheck(forms.start, forms.end)) {
      const hour = forms.start.split(':')[0];
      const time = `${hour}:59`;
      setForms({ ...forms, end: time });
      setRequired({ ...required, end: time });
    }
  }, [forms, required]);

  return (
    <form autoComplete="off" className={classes.root}>
      <div className={classes.titleBox}>
        <CreateIcon style={{ fontSize: 16, margin: '0 4px 0 0' }} />
        <Typography variant="subtitle2">{title(mode)} Form</Typography>
      </div>
      <TextField
        fullWidth
        label="タイトル"
        placeholder="タイトルを入力"
        size="small"
        name="title"
        className={classes.marginBottom}
        helperText={errors.title}
        error={validator.isError(errors.title)}
        value={forms.title}
        onChange={handleChange}
      />
      <dl className={classNames([classes.reset])}>
        {/* Date */}
        <div className={classes.formFlex}>
          <dt className={classNames([classes.formTitle, classes.reset])}>
            年/月/日
          </dt>
          <dd className={classNames([classes.marginBottom, classes.reset])}>
            <TextField
              type="date"
              name="date"
              value={forms.date}
              onChange={handleChange}
            />
          </dd>
        </div>
        {/* Start */}
        <div className={classes.formFlex}>
          <dt className={classNames([classes.formTitle, classes.reset])}>
            開始〜終了 時間
          </dt>
          <dd className={classNames([classes.marginBottom, classes.reset])}>
            <TextField
              type="time"
              name="start"
              helperText={errors.start}
              error={validator.isError(errors.start)}
              value={forms.start}
              onChange={handleChange}
            />
            <span className={classes.marginRight}>〜</span>
            <TextField
              type="time"
              name="end"
              helperText={errors.end}
              error={validator.isError(errors.end)}
              value={forms.end}
              onChange={handleChange}
            />
          </dd>
        </div>
        {/* Coment */}
        <div className={classNames([classes.formFlex])}>
          <dt
            className={classNames([classes.formTextAreaTitle, classes.reset])}
          >
            メモ
          </dt>
          <dd className={classes.reset}>
            <TextField
              className={classes.formTextArea}
              inputProps={{ style: { fontSize: 14, lineHeight: 1.3 } }}
              size="small"
              name="memo"
              multiline
              variant="outlined"
              rows={4}
              value={forms.memo}
              onChange={handleChange}
            />
          </dd>
        </div>
        {/* Delete Check */}
        {mode !== '0' ? (
          <div className={classNames([classes.marginBottom])}>
            <Checkbox
              size="small"
              classes={{ root: classes.formCheckBox }}
              name="delete"
              checked={forms.delete}
              onChange={handleCheckBox}
            />
            <span className={classes.deleteText}>データを削除する</span>
          </div>
        ) : (
          ''
        )}
        {/* Duplicate Error */}
        {mode === '0' && isDupError ? <ErrorMsg /> : ''}
        <div className={classes.buttonArea}>
          <Button
            aria-label={label(mode)}
            variant="contained"
            className={classNames([classes.okButton])}
            disableElevation
            disabled={!isButtonDisabled()}
            onClick={submitButton}
          >
            OK
          </Button>
          {isScreen !== true ? (
            <Button
              size="small"
              aria-label="cancel"
              variant="outlined"
              disableElevation
              onClick={close}
            >
              Cancel
            </Button>
          ) : (
            ''
          )}
        </div>
      </dl>
    </form>
  );
}

export default Form;
