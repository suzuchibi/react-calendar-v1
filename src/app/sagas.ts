import { put, call, takeEvery } from 'redux-saga/effects';
import { CallEffect, PutEffect } from 'redux-saga/effects';
import { Props as SagaProps } from './sagaActions';
import { Holidays } from '../features/calendar/types';
import GAPI from '../api/googleCalendarApi';

import config from '../config/app.json';
import {
  getYear,
  getMonth,
  getDate,
  startOfMonth,
  endOfMonth,
  add,
} from 'date-fns';

export interface ApiProps {
  type: string;
  payload: SagaProps;
}
export interface SuccessProps {
  type: string;
  payload: {
    mode: string;
    result: Array<Holidays>;
  };
}
export interface ErrorProps {
  type: string;
  payload: {
    mode: string;
  };
}
export type SagaTypes = CallEffect | PutEffect;

/**
 * SetDateText タイムゾーンのテキスト化
 * @param {Date | number} e
 * @return {String}
 */
const setTimeText = (e: Date | number) => {
  const pad = (t: number) => ('00' + t).slice(-2);
  return `${getYear(e)}-${pad(getMonth(e) + 1)}-${pad(getDate(e))}T00:00:00Z`;
};

/**
 * GetTimeZone 開始と終了の日付取得
 * @param {number} stamp
 * @param {String} mode
 * @return {Object<{start: String, end: String}>}
 */
const getTimeZone = (stamp: number, mode: String) => {
  switch (mode) {
    case 'cal':
      const s = startOfMonth(stamp);
      const e = endOfMonth(stamp);
      return {
        start: setTimeText(s),
        end: setTimeText(e),
      };
    case 'schedule':
      const dateAddSix = add(stamp, { days: 7 });
      return {
        start: setTimeText(stamp),
        end: setTimeText(dateAddSix),
      };
    case 'all':
      const sAll = startOfMonth(stamp);
      const end = endOfMonth(stamp);
      const endAddSix = add(end, { days: 6 });
      return {
        start: setTimeText(sAll),
        end: setTimeText(endAddSix),
      };
    default:
      return {
        start: '',
        end: '',
      };
  }
};

// ここからジェネレーター

/**
 * Success 成功した後のディスパッチ処理
 * @param {object} payload
 * @returns {void}
 */
export function* success({
  payload,
}: SuccessProps): Generator<PutEffect, void, SuccessProps> {
  const { mode, result } = payload;
  switch (mode) {
    case 'cal':
      yield put({ type: 'calendar/changeHolidays', payload: result });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      break;
    case 'schedule':
      yield put({ type: 'schedule/changeHolidays', payload: result });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
    case 'all':
      yield put({ type: 'calendar/changeHolidays', payload: result });
      yield put({ type: 'schedule/changeHolidays', payload: result });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
  }
}

/**
 * Error 失敗した後のディスパッチ処理
 * @param {object} payload modeの値が返る
 * @returns {void}
 */
export function* error({
  payload,
}: ErrorProps): Generator<PutEffect, void, string> {
  const { mode } = payload;
  switch (mode) {
    case 'cal':
      yield put({ type: 'calendar/changeHolidays', payload: [] });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      break;
    case 'schedule':
      yield put({ type: 'schedule/changeHolidays', payload: [] });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
    case 'all':
      yield put({ type: 'calendar/changeHolidays', payload: [] });
      yield put({ type: 'schedule/changeHolidays', payload: [] });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
  }
}

/**
 * Generator GoogleCalendarApi
 * @param {object} payload
 * @returns {void}
 */
export function* googleCalendarApiSaga({
  payload,
}: ApiProps): Generator<SagaTypes, void, any> {
  const { mode, stamp } = payload;
  const { start, end } = getTimeZone(stamp, mode);
  const param = `timeMin=${start}&timeMax=${end}&singleEvents=true`;
  const encode = encodeURIComponent(config.gapiCal.address);
  const path = `${config.gapiCal.gurl}${encode}/events?${param}`;
  try {
    // GoogleCalendarAPIコール
    const result = yield call(GAPI.start, path);
    yield put({
      type: 'app/googleCalendarApiSuccess',
      payload: { mode: mode, result: result },
    });
  } catch (err) {
    yield put({ type: 'snac/changeMsg', payload: err });
    yield put({ type: 'snac/changeOpen', payload: true });
    yield put({
      type: 'app/googleCalendarApiError',
      payload: { mode: mode },
    });
  }
}

export default function* rootSaga() {
  yield takeEvery('app/googleCalendarAPI', googleCalendarApiSaga);
  yield takeEvery('app/googleCalendarApiSuccess', success);
  yield takeEvery('app/googleCalendarApiError', error);
}
