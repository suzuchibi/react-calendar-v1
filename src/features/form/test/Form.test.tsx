import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Form from '../Form';

import { mockStore as store } from '../../../app/mockStore';

// Termial
// yarn test src/features/form/test/Form.test.ts

describe('<Form>', () => {
  it('スナップショット テスト', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Form />
      </Provider>
    );
    // snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
