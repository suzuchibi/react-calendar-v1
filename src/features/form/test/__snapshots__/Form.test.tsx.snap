// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> スナップショット テスト 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-titleBox-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="font-size: 16px; margin: 0px 4px 0px 0px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
        Add Form
      </h6>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-marginBottom-5 MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
        data-shrink="false"
      >
        タイトル
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
          name="title"
          placeholder="タイトルを入力"
          type="text"
          value=""
        />
      </div>
    </div>
    <dl
      class="makeStyles-reset-3"
    >
      <div
        class="makeStyles-formFlex-6"
      >
        <dt
          class="makeStyles-formTitle-7 makeStyles-reset-3"
        >
          年/月/日
        </dt>
        <dd
          class="makeStyles-marginBottom-5 makeStyles-reset-3"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="date"
                type="date"
                value="2021-07-12"
              />
            </div>
          </div>
        </dd>
      </div>
      <div
        class="makeStyles-formFlex-6"
      >
        <dt
          class="makeStyles-formTitle-7 makeStyles-reset-3"
        >
          開始〜終了 時間
        </dt>
        <dd
          class="makeStyles-marginBottom-5 makeStyles-reset-3"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="start"
                type="time"
                value="17:00"
              />
            </div>
          </div>
          <span
            class="makeStyles-marginRight-4"
          >
            〜
          </span>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="end"
                type="time"
                value="18:00"
              />
            </div>
          </div>
        </dd>
      </div>
      <div
        class="makeStyles-formFlex-6"
      >
        <dt
          class="makeStyles-formTextAreaTitle-9 makeStyles-reset-3"
        >
          メモ
        </dt>
        <dd
          class="makeStyles-reset-3"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-formTextArea-10"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                name="memo"
                rows="4"
                style="font-size: 14px; line-height: 1.3;"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-17"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </dd>
      </div>
      <div
        class="makeStyles-buttonArea-14"
      >
        <button
          aria-label="add"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-okButton-15 MuiButton-disableElevation Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            OK
          </span>
        </button>
        <button
          aria-label="cancel"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </dl>
  </form>
</DocumentFragment>
`;
